.github/workflows/internal-archive-release.yml

```diff
@@ -32,13 +46,47 @@ defaults:
     shell: bash
 
 env:
-  PYTHON_TARGET_VERSION: 3.11
+  PYTHON_TARGET_VERSION: 3.8
   NOTIFICATION_PREFIX: "[Internal Archive Release]"
+  TEMP_PROFILE_NAME: "temp_aws_profile"
+  HATCH_ADAPTERS: '["postgres"]' # Must be valid JSON
+
+
 
 jobs:
+  initial-setup:
+    runs-on: ubuntu-latest
+    outputs:
+      is_hatch_adapter: ${{ steps.check_if_adapter_builds_with_hatch.outputs.is_hatch_adapter }}
+      is_spark_build: ${{ steps.check_if_this_is_a_spark_build.outputs.is_spark_build }}
+    steps:
+      - name: "Check if dbms_name is contained in HATCH_ADAPTERS"
+        id: "check_if_adapter_builds_with_hatch"
+        run: |
+          echo "HATCH_ADAPTERS='${{ env.HATCH_ADAPTERS }}'" >> $GITHUB_ENV
+          ADAPTERS="$(jq -r '.' <<< "${HATCH_ADAPTERS}")"
+          echo "$ADAPTERS"
+          DBMS_NAME="${{ inputs.dbms_name }}"
+          if jq -e --arg dbms "$DBMS_NAME" '.[] | select(. == $dbms)' <<< ${ADAPTERS}; then
```

Comment:
This feel overly complicated.  If HATCH_ADAPTERS is just a space delimited list you can just check that the dbms is in the list:

```
          if [[ " ${{ env.HATCH_ADAPTERS}} " =~ " ${{ inputs.dbms_name }} " ]]; then
              echo "is_hatch_adapter=true" >> $GITHUB_OUTPUT
          else
              echo "is_hatch_adapter=false" >> $GITHUB_OUTPUT
          fi
```

I'm also fine with leaving this for now and simplifying it at a future date.  Not a blocker.

Reply:
Let me address in the upcoming round 2 which will add third party support. But geez okay this is news to me. For what it's worth, I went with a comment delimited list as the goal but I realize now it's more bash idiomatic to use the space delimited list. It's rare I get to up my bash fu these days -- thanks for the prompt! 🙇‍♀️

Reply:
```
" ${{ env.HATCH_ADAPTERS}} " =~ " ${{ inputs.dbms_name }} "
```

This is how you say `inputs.dbms_name` is in `env.HATCH_ADAPTERS` in GHA?